src/rdb.c

```diff
@@ -3053,7 +3052,8 @@ int rdbLoadRioWithLoadingCtx(rio *rdb, int rdbflags, rdbSaveInfo *rsi, rdbLoadin
     }
     rdbver = atoi(buf + 6);
     if (rdbver < 1 ||
-        (rdbver >= RDB_FOREIGN_VERSION_MIN && !is_valkey_magic) ||
+        (rdbver >= RDB_FOREIGN_VERSION_MIN && rdbver <= RDB_FOREIGN_VERSION_MAX) ||
+        (rdbver > RDB_FOREIGN_VERSION_MAX && !is_valkey_magic) ||
         (rdbver > RDB_VERSION && server.rdb_version_check == RDB_VERSION_CHECK_STRICT)) {
         serverLog(LL_WARNING, "Can't handle RDB format version %d", rdbver);
```

Comment:
We should probably extend this to include whether or not it's the Redis magic as well?

Reply:
@madolson to which one of the check conditions do you refer to? The [12-79] or just validate that below 12 we have the redis magic?

Reply:
@madolson done. want to take a look?

Reply:
Apparently not :D

Reply:
Since we don't reuse the same numbers as with the different magics, it will never be ambiguous (except for hand-hacked RDB files with wrong magic/number).

Reply:
For us yes, not for end users. Ideally we should print out that this is a Redis RDB and not a valkey RDB.